internal/provisioner/kops_provisioner.go

```diff
@@ -878,11 +900,21 @@ func (provisioner *KopsProvisioner) UpdateClusterInstallation(cluster *model.Clu
 
 	version := translateMattermostVersion(installation.Version)
 	if cr.Spec.Version == version {
-		logger.Debugf("Cluster installation already on version %s", version)
-		return nil
+		logger.Infof("Cluster installation already on version %s", version)
 	}
 	cr.Spec.Version = version
 
+	cr.Spec.MattermostLicenseSecret = ""
+	var secretName string
+	if installation.License != "" {
+		secretName = fmt.Sprintf("%s-license", name)
+		_, err = k8sClient.CreateSecret(clusterInstallation.Namespace, secretName, "license", installation.License)
```

Comment:
Correct me if I am wrong, but this won't update the secret if there was a previously created secret right? The function will return with no error when it finds the previous secret and it won't be updated.

Reply:
good catch!
so we need to check if the secret has the same content if is different then update the secret.

the case can be: an existing installation have a license but that expires and we need to update the license.

I will refactor this part to make the appropriate checks

thanks for your review @gabrieljackson

Reply:
Sure thing!

We may also want to make logic to remove a license if the value is set to `""` although that is less pressing than the update issue I noted above.

Reply:
done @gabrieljackson